DOCS-942: Add MoveOnGlobe() to the Motion Service docs (#1570)
viamrobotics/docs: docs/services/motion/_index.md

----------- | -----------
[`Move`](#move) | Move multiple components in a coordinated way to achieve a desired motion.
[`MoveSingleComponent`](#movesinglecomponent) | Move a single component "manually."
[`GetPose`](#getpose) | Get the current location and orientation of a component as a `Pose`.

> [`MoveOnMap`](#moveonmap) | Move a component to a `Pose` in respect to the origin of a [SLAM](/services/slam/) map.

Author: ```suggestion
[`MoveOnMap`](#moveonmap) | Move a [base](/components/base/) component to a `Pose` in respect to the origin of a [SLAM](/services/slam/) map.
```

> [`MoveOnGlobe`] | Move a component to a destination GPS point. Use a [Movement Sensor](/components/movement-sensor/) to measure the robot's GPS coordinates.

Reviewer: ```suggestion
[`MoveOnGlobe`](/services/motion/#moveonglobe) | Move a component to a destination GPS point. Use a [Movement Sensor](/components/movement-sensor/) to measure the robot's GPS coordinates.
```

> [`MoveOnGlobe`] | Move a component to a destination GPS point. Use a [Movement Sensor](/components/movement-sensor/) to measure the robot's GPS coordinates.

Author: ```suggestion
[`MoveOnGlobe`] | Move a [base](/components/base/) component to a destination GPS point. Use a [Movement Sensor](/components/movement-sensor/) to measure the robot's GPS coordinates.
```

{{% /tab %}}
{{< /tabs >}}

> ### MoveOnMap

Author: I moved this down so that the methods would align with the order shown in the table, and it makes sense comprehensively to group the endpoints that produce motion together

- `obstacles` [([]*spatialmath.GeoObstacle)](https://pkg.go.dev/go.viam.com/rdk/spatialmath#GeoObstacle): Obstacles to consider when planning the motion of the component, with each represented as a `GeoObstacle`. <ul><li> Default: `None` </li></ul>
- `linear_meters_per_sec` [(float64)](https://pkg.go.dev/builtin#float64): The linear velocity in meters per second to target when moving the component. <ul><li> Default: `0.3` </li></ul>
- `angular_deg_per_sec` [(float64)](https://pkg.go.dev/builtin#float64): The [angular velocity](https://en.wikipedia.org/wiki/Angular_velocity) in degrees per second to target when turning the component. <ul><li> Default: `60.0` </li></ul>
- `extra` [(map\[string\]interface{})](https://go.dev/blog/maps): Extra options to pass to the underlying RPC call.

**Returns:**

- [(bool)](https://pkg.go.dev/builtin#bool): Whether the request to `MoveOnGlobe` was successful.
- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.

For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk/services/motion#Service).

```go {class="line-numbers linkable-line-numbers"}
motionService, err := motion.FromRobot(robot, "builtin")

> // Define a destination Point at the geographic location [0, 0]

Author: ```suggestion
// Define a destination Point at the GPS coordinates [0, 0]
```

{{% /tab %}}
{{< /tabs >}}

### MoveOnMap

> Move a [base](/component/base/) component to a [`Pose`](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.Pose) in respect to the origin of a [SLAM](/services/slam/) map.

Author: ```suggestion
Move a [base](/components/base/) component to a [`Pose`](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.Pose) in respect to the origin of a [SLAM](/services/slam/) map.
```

{{% /tab %}}
{{< /tabs >}}

### MoveOnMap

> Move a component to a [`Pose`](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.Pose) in respect to the origin of a [SLAM](/services/slam/) map.

Author: ```suggestion
Move a [base](/component/base/) component to a [`Pose`](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.Pose) in respect to the origin of a [SLAM](/services/slam/) map.
```

{{< tabs >}}
{{% tab name="Python" %}}

**Parameters:**

> - `component_name` ([ResourceName](https://python.viam.dev/autoapi/viam/gen/common/v1/common_pb2/index.html#viam.gen.common.v1.common_pb2.ResourceName)): The `"name"` of the component to move.

Author: ```suggestion
- `component_name` ([ResourceName](https://python.viam.dev/autoapi/viam/gen/common/v1/common_pb2/index.html#viam.gen.common.v1.common_pb2.ResourceName)): The `"name"` of the base you want to move.
```

**Returns:**

- [(bool)](https://docs.python.org/3/library/stdtypes.html#bltin-boolean-values): Whether the request to `MoveOnMap` was successful.

For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/services/motion/index.html#viam.services.motion.MotionClient.move_on_map).

```python {class="line-numbers linkable-line-numbers"}
motion = MotionClient.from_robot(robot=robot, name="builtin")

# Define a destination pose with respect to the origin of the map from the SLAM service "my_slam_service"
my_pose = Pose(y=10)

# Move the base component "my_base" to the destination pose of Y=10, a location of (0, 10, 0) in respect to the origin of the map
success = await motion.move_on_map(component_name="my_base", destination=my_pose, slam_service_name="my_slam_service")
```

{{% /tab %}}
{{% tab name="Go" %}}

**Parameters:**

- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.

> - `componentName` [(resource.Name)](https://pkg.go.dev/go.viam.com/rdk/resource#Name): The `"name"` of the component to move.

Author: ```suggestion
- `componentName` [(resource.Name)](https://pkg.go.dev/go.viam.com/rdk/resource#Name): The `"name"` of the base to move.
```

- `slamName` [(resource.Name)](https://pkg.go.dev/go.viam.com/rdk/resource#Name): The `"name"` of the [SLAM Service](/services/slam/) from which the SLAM map is requested.
- `extra` [(map\[string\]interface{})](https://go.dev/blog/maps): Extra options to pass to the underlying RPC call.

**Returns:**

- [(bool)](https://pkg.go.dev/builtin#bool): Whether the request to `MoveOnMap` was successful.
- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.

For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk@v0.2.50/services/motion#Service).

```go {class="line-numbers linkable-line-numbers"}
motionService, err := motion.FromRobot(robot, "builtin")

// Define a destination Pose with respect to the origin of the map from the SLAM service "my_slam_service"
myPose := spatialmath.NewPoseFromPoint(r3.Vector{Y: 10})

// Move the base component "my_base" to the destination pose of Y=10, a location of (0, 10, 0) in respect to the origin of the map
success, err := motionService.MoveOnMap(context.Background(), "my_base", myPose, "my_slam_service", nil)
```

{{% /tab %}}
{{< /tabs >}}

### MoveOnGlobe

> Move a component to a destination GPS point, represented in geographic notation *(latitude, longitude)*.

Author: ```suggestion
Move a [base](/components/base/) component to a destination GPS point, represented in geographic notation *(latitude, longitude)*.
```

Use a [Movement Sensor](/components/movement-sensor/) to check the location of the robot.

{{< alert title="Usage" color="tip" >}}

> `MoveOnGlobe()` is intended for use with the [Navigation Service](/services/navigation), providing autonomous GPS navigation for rover [bases](/components/base/).

Author: ```suggestion
`MoveOnGlobe()` is intended for use with the [Navigation Service](/services/navigation/), providing autonomous GPS navigation for rover [bases](/components/base/).
```

To use `MoveOnGlobe()`, your movement sensor must be able to measure the GPS location and orientation of the robot.

Make sure the [movement sensor](/components/movement-sensor/) you use supports usage of the following methods in its {{< glossary_tooltip term_id="model" text="model's" >}} implementation of the [Movement Sensor API](/components/movement-sensor/#api).

- It must support `GetPosition()` to report the robot's current GPS location.
- It must **also** support **either** `GetCompassHeading()` or `GetOrientation()` to report which way the robot is facing.
- If your movement sensor provides multiple methods, your robot will default to using the values returned by `GetCompassHeading()`.

{{< /alert >}}

{{< alert title="Stability Notice" color="alert" >}}

> This method is experimental.

Reviewer 2: MoveOnGlobe is not experimental. Only specifying "heading" when the minimum turning radius is greater than zero is experimental.

Author: Ok awesome. Making this clear

Specifying `obstacles` in a request to `MoveOnGlobe()` will cause an error if you configure a `"translation"` in the `"geometries"` of any of the `GeoObstacle` objects.

> The translation feature is not yet supported by the [Navigation Service](/services/navigation/).

Reviewer 2: We do not plan to support translation, since you can already specify the location using the latitude and longitude.

> The translation feature is not yet supported by the [Navigation Service](/services/navigation/).

Author: ```suggestion
Translation in obstacles is not supported by the [Navigation Service](/services/navigation/).
```

> Specifying `heading` in a request to `MoveOnGlobe` is not currently recommended if the minimum turning radius of your component is greater than zero, as this combination may cause high latency in the [motion planning algorithms](/services/motion/algorithms/).

Author: ```suggestion
```

**Returns:**

- [(bool)](https://docs.python.org/3/library/stdtypes.html#bltin-boolean-values): Whether the request to `MoveOnGlobe` was successful.

For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/services/motion/index.html#viam.services.motion.MotionClient.move_on_globe).

```python {class="line-numbers linkable-line-numbers"}
motion = MotionClient.from_robot(robot=robot, name="builtin")

#  Define a destination GeoPoint at the GPS coordinates [0, 0]
my_destination = movement_sensor.GeoPoint(latitude=0, longitude=0)

# Move the base component "my_base" to the designated geographic location, as reported by the movement sensor "my_movement_sensor"
success = await motion.move_on_globe(component_name="my_base", destination=my_destination, movement_sensor_name="my_movement_sensor")
```

{{% /tab %}}
{{% tab name="Go" %}}

**Parameters:**

- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.

> - `componentName` [(resource.Name)](https://pkg.go.dev/go.viam.com/rdk/resource#Name): The `"name"` of the component to move.

Author: ```suggestion
- `componentName` [(resource.Name)](https://pkg.go.dev/go.viam.com/rdk/resource#Name): The `"name"` of the base to move.
```
